packages/office-ui-fabric-react/src/components/ResizeGroup/ResizeGroup.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import {
+  css,
+  BaseComponent
+} from '../../Utilities';
+import { IResizeGroupProps, IResizeGroup } from './ResizeGroup.Props';
+import styles = require('./ResizeGroup.scss');
+
+export interface IResizeGroupState {
+  renderedData?: any;
+  measuredData: any;
+  shouldMeasure?: boolean;
+}
+
+export class ResizeGroup extends BaseComponent<IResizeGroupProps, IResizeGroupState> implements IResizeGroup {
+
+  public static defaultProps = {
+    data: {}
+  };
+
+  private _root: HTMLElement;
+  private _measured: HTMLElement;
+
+  constructor(props: IResizeGroupProps) {
+    super(props);
+    this.state = {
+      shouldMeasure: true,
+      renderedData: null,
+      measuredData: { ...this.props.data },
+    };
+  }
+
+  public componentDidMount() {
+    this._measureItems();
+    this._events.on(window, 'resize', this._onResize);
+  }
+
+  public measure = () => {
+    this._onResize();
+  }
+
+  public render() {
+    const { onRenderData, onReduceData, data } = this.props;
+    let { shouldMeasure, renderedData, measuredData } = this.state;
+
+    if (Object.keys(data).length === 0) {
+      return null;
+    }
+
+    return (
+      <div ref={ this._resolveRef('_root') }>
+        { shouldMeasure && (
+          <div className={ css(styles.measured) } ref={ this._resolveRef('_measured') }>
+            { onRenderData(measuredData) }
+          </div>
+        ) }
+
+        { renderedData && onRenderData(renderedData) }
```

Comment:
it always needs to start at the same place so that the onReduceItems function functions the same. i.e. we don't  have a reduceItems and a increaseItems functions, which means we always need to start from the props state, and reduce until it fits.